src/Symfony/Component/Yaml/Inline.php

```diff
@@ -504,6 +504,11 @@ private static function parseMapping($mapping, $flags, &$i = 0, $references = []
             $isKeyQuoted = \in_array($mapping[$i], ['"', "'"], true);
             $key = self::parseScalar($mapping, $flags, [':', ' '], $i, false, [], true);
 
+            if ('!php/const' === $key) {
```

Comment:
Shouldn't we check for `self::$constantSupport` first? /cc @xabbuh

Reply:
I think this is fine because `evaluateScalar()` (which is called now) does the check.